inspire_dojson/cds/rules.py

```diff
@@ -0,0 +1,443 @@
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import re
+from itertools import chain
+
+import pycountry
+import six
+
+from idutils import is_arxiv
+from dojson import utils
+
+from six.moves import urllib
+
+from inspire_utils.helpers import force_list
+
+from .model import cds2hep_marc
+from ..utils import force_single_element
+
+CATEGORIES = {
+    'General Relativity and Cosmology': 'Gravitation and Cosmology',
+    'General Theoretical Physics': 'General Physics',
+    'Detectors and Experimental Techniques': 'Instrumentation',
+    'Engineering': 'Instrumentation',
+    'Accelerators and Storage Rings': 'Accelerators',
+    'Computing and Computers': 'Computing',
+    'Mathematical Physics and Mathematics': 'Math and Math Physics',
+    'Astrophysics and Astronomy': 'Astrophysics',
+    'Condensed Matter': 'General Physics',
+    'General Theoretical Physics': 'General Physics',
+    'Physics in General': 'General Physics',
+    'Other': 'Other',
+    'Chemical Physics and Chemistry': 'Other',
+    'Information Transfer and Management': 'Other',
+    'Commerce, Economics, Social Science': 'Other',
+    'Biography, Geography, History': 'Other',
+    'Science in General': 'Other',
+    'Particle Physics - Experiment': 'Experiment-HEP',
+    'Particle Physics - Phenomenology': 'Phenomenology-HEP',
+    'Particle Physics - Theory': 'Theory-HEP',
+    'Particle Physics - Lattice': 'Lattice',
+    'Nuclear Physics - Experiment': 'Experiment-Nucl',
+    'Nuclear Physics - Theory': 'Theory-Nucl',
+}
+
+
+EXPERIMENTS = {
+    ('CERN LEP', 'L3'): 'CERN-LEP-L3',
+    ('CERN LEP', 'OPAL'): 'CERN-LEP-OPAL',
+    ('CERN LHC', 'ALICE'): 'CERN-LHC-ALICE',
+    ('CERN LHC', 'ATLAS'): 'CERN-LHC-ATLAS',
+    ('CERN LHC', 'CMS'): 'CERN-LHC-CMS',
+    ('CERN LHC', 'LHCb'): 'CERN-LHC-LHCb',
+    ('CERN LHC', 'LHCf'): 'CERN-LHC-LHCf',
+    ('CERN LHC', 'MoEDAL'): 'CERN-LHC-MoEDAL',
+    ('CERN LHC', 'TOTEM'): 'CERN-LHC-TOTEM',
+    ('CERN PS', 'nTOF'): 'CERN-nTOF',
+    ('CERN SPS', 'ICARUS CNGS2'): 'ICARUS',
+    ('CERN SPS', 'OPERA CNGS1'): 'OPERA',
+    ('DESY HERA', 'ZEUS'): 'DESY-HERA-ZEUS',
+}
+
+
+RE_IDS = re.compile(r'\((?P<schema>.*?)\)(?P<id>.*)')
+
+
+def add_source(field, source='CDS'):
+    if not field.get('9'):
+        field['9'] = source
+
+
+def vanilla_dict(god):
+    return {k: v for (k, v) in six.iteritems(god) if k != '__order__'}
+
+
+@cds2hep_marc.over('0247_', '^0247.')
+@utils.for_each_value
+def persistent_identifiers(self, key, value):
+    add_source(value)
+
+    return vanilla_dict(value)
+
+
+@cds2hep_marc.over('035__', '^035..')
+@utils.for_each_value
+def external_sytem_identifiers(self, key, value):
+    ignored = {'cercer', 'inspire', 'xx', 'cern annual report', 'cmscms', 'wai01', 'spires'}
+    if not any(val.lower() in ignored for val in chain(force_list(value.get('9')), force_list(value.get('a')))):
+        return vanilla_dict(value)
+
+
+@cds2hep_marc.over('037__', '^037..', '^088..')
```

Comment:
Is this equivalent to `@cds2hep_marc.over('037__', '(^037..|^088..)')` ?

Reply:
[yes](https://github.com/inveniosoftware/dojson/blob/9ac1f416b112da04bf79634b600f07e6fb091a11/dojson/overdo.py#L87-L94)